src/coreComponents/constitutive/solid/CamClay.hpp

```diff
@@ -0,0 +1,611 @@
+      real64 eps_v_trial;
+
+      twoInvariant::stressDecomposition( stress,
+                                         oldP,
+                                         oldQ,
+                                         oldDeviator );
+    
+    
+
+    // Recover elastic strains from the previous step, based on stress from the previous step
+    // [Note: in order to minimize data transfer, we are not storing and passing elastic strains]
+    
+    real64 oldElasticStrainVol = std::log(oldP/p0) * Cr * (-1.0) + eps_v0 ;
+    real64 oldElasticStrainDev = oldQ/3./mu;
+   
+    // Now recover the old strain tensor from the strain invariants.
+    // Note that we need the deviatoric direction (n-hat) from the previous step.
+    
+    twoInvariant::strainRecomposition( oldElasticStrainVol,
+                                       oldElasticStrainDev,
+                                       oldDeviator,
+                                       oldStrainElastic );
+    
+    // elastic predictor (assume strainIncrement is all elastic)
+    // TODO: define ElasticIsotropicPressureUpdates to call here
+    //ElasticIsotropicUpdates::smallStrainUpdate( k, q, strainIncrement, stress, stiffness );
+
+    for(localIndex i=0; i<6; ++i)
+   {
+     strainElasticTrial[i] = oldStrainElastic[i] + strainIncrement[i];
+   }
+  // two-invariant decomposition of trial elastic strain
+    
+    twoInvariant::strainDecomposition( strainElasticTrial,
+                                       eps_v_trial,
+                                       eps_s_trial,
+                                      deviator);
+  
+    // Calculate trial mean and deviatoric stress
+    
+    real64 trialP = p0 * std::exp(-1./Cr* (eps_v_trial-eps_v0));
+    real64 trialQ = 3. * mu * eps_s_trial;
+    
+    twoInvariant::stressRecomposition( trialP,
+                                       trialQ,
+                                       deviator,
+                                      stress );
+    
+    // set stiffness to elastic predictor
+    
+    bulkModulus = -trialP/Cr ;
+    real64 lame = bulkModulus - 2./3. * mu;
+
+    for(localIndex i=0; i<6; ++i)
+    {
+      for(localIndex j=0; j<6; ++j)
+      {
+        stiffness[i][j] = 0;
+      }
+    }
+    
+    stiffness[0][0] = lame + 2.*mu;
+    stiffness[0][1] = lame;
+    stiffness[0][2] = lame;
+
+    stiffness[1][0] = lame;
+    stiffness[1][1] = lame + 2.*mu;
+    stiffness[1][2] = lame;
+
+    stiffness[2][0] = lame;
+    stiffness[2][1] = lame;
+    stiffness[2][2] = lame + 2.*mu;
+
+    stiffness[3][3] = mu;
+    stiffness[4][4] = mu;
+    stiffness[5][5] = mu;
+    
+    // check yield function F <= 0
+    real64 yield = trialQ*trialQ/(M*M)- alpha*alpha*trialP *(2*alpha/(alpha+1)*pc-trialP)+alpha*alpha*(alpha-1)/(alpha+1)* pc*pc;
+
+  if( yield < 1e-9 ) // elasticity
+  {
+    return;
+  }
+
+  // else, plasticity (trial stress point lies outside yield surface)
+  std::cout << "plastic " <<  "\n " << std::endl;
```

Comment:
Remove when done debugging.  This is why the CUDA checks are failing.

Reply:
Yes, sure. I will remove that.